core/app/components/forms/actions.ts

```diff
@@ -86,17 +92,17 @@ export const deleteFile = defineServerAction(async function deleteFile({
 			if (mode === "edit") {
 				// delete pubvalue if it exists
 
+				// TODO: should we remove the pubvalue if it's an empty array?
 				await autoRevalidate(
 					trx
 						.updateTable("pub_values")
 						.where("pub_values.pubId", "=", pubId)
-						.where((eb) =>
-							eb.exists((eb) =>
-								eb
-									.selectFrom("pub_fields")
-									.select("slug")
-									.where("slug", "=", fieldSlug)
-							)
+						.where("pub_values.fieldId", "=", (eb) =>
+							eb
+								.selectFrom("pub_fields")
+								.select("pub_fields.id")
+								.where("pub_fields.slug", "=", fieldSlug)
+								.limit(1)
```

Comment:
previous iteration would delete _all_ files on all fields of the pub if they have the same url, which isn't what we want